recipes/libsndfile/all/conanfile.py

```diff
@@ -0,0 +1,104 @@
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://www.mega-nerd.com/libsndfile"
+    license = "LGPL-2.1"
+    generators = 'pkg_config'
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_alsa": [True, False],
+        "with_sqlite": [True, False],
+        "with_external_libs": [True, False]}
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_alsa": True,
+        "with_sqlite": True,
+        "with_external_libs": True,
+        "libalsa:shared": True}
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def requirements(self):
+        if self.options.with_alsa:
+            self.requires("libalsa/1.1.9")
+        if self.options.with_sqlite:
+            self.requires("sqlite3/3.29.0")
+        if self.options.with_external_libs:
+            self.requires("flac/1.3.3")
+            self.requires("ogg/1.3.4")
+            self.requires("vorbis/1.3.6")
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.compiler == "Visual Studio":
+            raise ConanInvalidConfiguration("The project libsndfile can not be built by Visual Studio.")
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        del self.settings.compiler.stdcpp
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_autotools(self):
+        if not self._autotools:
+            self._autotools = AutoToolsBuildEnvironment(self)
+            args = ["--without-caps"]
+            if not self.options.with_alsa:
+                args.append("--disable-alsa")
+            if not self.options.with_sqlite:
+                args.append("--disable-sqlite")
+            if not self.options.with_external_libs:
+                args.append("--disable-external-libs")
+            if self.options.shared:
+                args.extend(['--enable-shared=yes', '--enable-static=no'])
+            else:
+                args.extend(['--enable-shared=no', '--enable-static=yes'])
+            self._autotools.configure(args=args, configure_dir=self._source_subfolder)
+        return self._autotools
+
+    def build(self):
+        if self.options.with_external_libs:
+            shutil.copyfile('vorbis.pc', 'vorbisenc.pc')
+        with tools.environment_append(RunEnvironment(self).vars):
+            try:
+                autotools = self._configure_autotools()
+                autotools.make()
+            except:
+                self.output.info(open('config.log', errors='backslashreplace').read())
+                raise
```

Comment:
```suggestion
        autotools = self._configure_autotools()
        autotools.make()
```

Reply:
RunEnvironment has to be used here, to avoid this error https://github.com/conan-io/conan-center-index/pull/1103#issuecomment-599212885

Reply:
I think that error is caused because you added `"libalsa:shared": True` to `default_options`.
It fails because libalsa.so cannot be found when trying to run a test executable.

Is `libalsa:shared=True` required?
The `RunEnvironment` should probably stay, because consumers might always select a shared libalsa.

Reply:
I cannot remember why `libalsa:shared=True` is there, but it seems to pass without it. Thanks for catching this !

Reply:
Thanks for showing me this function.
I think some recipes will stop working when their dependencies are switched from static to shared libraries.
@uilianries What do you think?